### PR TITLE
constexpr and member init changes.

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/datasets.cpp [hunk 1]
```diff
@@ -39,409 +39,352 @@
...
-                "the ARM with the IPTC and NAA organizations."),
-                false, false, 2, 2, Exiv2::unsignedShort, IptcDataSets::envelope, ""),
-        DataSet(IptcDataSets::ARMVersion, "ARMVersion", N_("ARM Version"),
-                N_("This tag consisting of a binary number representing the particular "
-                "version of the ARM specified by tag <ARMId>."),
-                false, false, 2, 2, Exiv2::unsignedShort, IptcDataSets::envelope, ""),
-        DataSet(0xffff, "(Invalid)", "(Invalid)",
-                "(Invalid)", false, false, 0, 0, Exiv2::unsignedShort, IptcDataSets::envelope, "")
+    static constexpr DataSet envelopeRecord[] = {
+        {IptcDataSets::ModelVersion, "ModelVersion", N_("Model Version"),
```
Comment (piponazo) [inline src/datasets.cpp:49]: Could you please revert the formatting changes here and just keep the 1-line change about `static constexpr` ?

File: src/datasets.cpp [hunk 2]
```diff
@@ -39,409 +39,352 @@
...
-                N_("A binary number representing the particular version of the "
-                "object data preview file format specified in tag <PreviewFormat>."),
-                false, false, 2, 2, Exiv2::unsignedShort, IptcDataSets::application2, ""),
-        DataSet(IptcDataSets::Preview, "Preview", N_("Preview Data"),
-                N_("Binary image preview data."),
-                false, false, 0, 256000, Exiv2::undefined, IptcDataSets::application2, ""),
-        DataSet(0xffff, "(Invalid)", N_("(Invalid)"),
-                N_("(Invalid)"),
-                false, false, 0, 0, Exiv2::unsignedShort, IptcDataSets::application2, "")
+    static constexpr DataSet application2Record[] = {
```
Comment (piponazo) [inline src/datasets.cpp:132]: Could you please revert the formatting changes here and just keep the 1-line change about `static constexpr` ?

File: src/properties.cpp [hunk 3]
```diff
@@ -95,61 +95,70 @@ namespace Exiv2 {
...
-        { "http://www.metadataworkinggroup.com/schemas/keywords/","mwg-kw", xmpMWGKeywordInfo, N_("Metadata Working Group Keywords schema")     },
-        { "http://www.video/",                             "video",          xmpVideoInfo,     N_("XMP Extended Video schema")                   },
-        { "http://www.audio/",                             "audio",          xmpAudioInfo,     N_("XMP Extended Audio schema")                   },
-        { "http://rs.tdwg.org/dwc/index.htm",             "dwc",            xmpDwCInfo,       N_("XMP Darwin Core schema")     		            },
-        { "http://purl.org/dc/terms/",                    "dcterms",        xmpDctermsInfo,   N_("Qualified Dublin Core schema")                }, // Note: used as properties under dwc:record
-        { "http://ns.acdsee.com/iptc/1.0/",               "acdsee",         xmpAcdseeInfo,    N_("ACDSee XMP schema")                           },
-        { "http://ns.google.com/photos/1.0/panorama/",    "GPano",          xmpGPanoInfo,     N_("Google Photo Sphere XMP schema")              },
-
+    static constexpr XmpNsInfo xmpNsInfo[] = {
```
Comment (piponazo) [inline src/properties.cpp:98]: Could you please revert the formatting changes here and just keep the 1-line change about `static constexpr` ?

**Review by piponazo — CHANGES_REQUESTED** [review]
Thanks @neheb for your work.

As commented in previous PRs, let's be conservative about formatting changes at the moment (until we have a discussion about the clang-formatting options). If you remove the big formatting changes in the static tables, I will be happy to approve and merge the changes proposed here.

**Comment by neheb** [comment]
Annoying.

I'm adding [[gnu::unused]] which according to the standard, is allowed to be present even if the compiler does not support it.

This should not be an error under WIndows.

**Review by piponazo — DISMISSED** [review]
Thanks for your endless efforts in modernizing the codebase! 💪

**Review by piponazo — DISMISSED** [review]

**Review by piponazo — APPROVED** [review]

**Pull request merged**